offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/OfferTemplate.groovy

```diff
@@ -0,0 +1,370 @@
+                tableBody.append(discountPosition.outerHtml())
+                position++
+            }
+        }
+        // set the footer
+        String netCosts = Currency.format(offer.getNetDataGeneration())
+        dataGenerationTable.select("> tfoot .costs").first().text(netCosts)
+    }
+    private static void fillDataAnalysisItems(Document document, OfferContent offer) {
+        Element dataAnalysisTable = document.getElementById("data-analysis-items")
+        // remove table if no items are available
+        if (offer.getDataAnalysisItems().isEmpty()) {
+            dataAnalysisTable.remove()
+            return
+        }
+        // clear the table
+        dataAnalysisTable.select("tbody > tr").each {element ->
+            element.remove()
+        }
+        // fill the table
+        int position = 1
+        Element tableBody = dataAnalysisTable.selectFirst("tbody")
+        for (OfferItem offerItem in offer.getDataAnalysisItems()) {
+            OfferPosition offerPosition = OfferPosition.createProductItem(position, offerItem)
+            tableBody.append(offerPosition.outerHtml())
+            position++
+            if (offerItem.getQuantityDiscount() > 0) {
+                OfferPosition discountPosition = OfferPosition.createDiscount(position, position - 1, offerItem)
+                tableBody.append(discountPosition.outerHtml())
+                position++
+            }
+        }
+        // set the footer
+        String netCosts = Currency.format(offer.getNetDataAnalysis())
+        dataAnalysisTable.select(" > tfoot .costs").first().text(netCosts)
+    }
+    private static void fillDataManagementItems(Document document, OfferContent offer) {
+        Element dataManagementTable = document.getElementById("data-management-items")
+        // remove table if no items are available
+        if (offer.getDataManagementItems().isEmpty()) {
+            dataManagementTable.remove()
+            return
+        }
+        // clear the table
+        dataManagementTable.select("tbody > tr").each {element ->
+            element.remove()
+        }
+        // fill the table
+        int position = 1
+        Element tableBody = dataManagementTable.selectFirst("tbody")
+        for (OfferItem offerItem in offer.getDataManagementItems()) {
+            OfferPosition offerPosition = OfferPosition.createProductItem(position, offerItem)
+            tableBody.append(offerPosition.outerHtml())
+            position++
+            if (offerItem.getQuantityDiscount() > 0) {
+                OfferPosition discountPosition = OfferPosition.createDiscount(position, position - 1, offerItem)
+                tableBody.append(discountPosition.outerHtml())
+                position++
+            }
+        }
+        // set the footer
+        String netCosts = Currency.format(offer.getNetPMandDS())
+        document.select("#data-management-items > tfoot .costs").first().text(netCosts)
```

Comment:
why is the css style here different than for the other tables?

Reply:
good catch.
```suggestion
        dataManagementTable.select(" > tfoot .costs").first().text(netCosts)
```